.github/workflows/workflow_inference.yml

```diff
@@ -95,23 +99,31 @@ jobs:
             DF_SUFFIX=".cpu_and_deepspeed"
           fi
           TARGET=${{steps.target.outputs.target}}
-          docker build ./ --build-arg CACHEBUST=1 --build-arg http_proxy=${{ inputs.http_proxy }} --build-arg https_proxy=${{ inputs.https_proxy }} -f dev/docker/Dockerfile${DF_SUFFIX} -t ${TARGET}:latest && yes | docker container prune && yes
-          docker image prune -f
+
+          source docker.sh
+          build_and_prune TARGET=${TARGET} DF_SUFFIX=${DF_SUFFIX} USE_PROXY=${"True"}
 
       - name: Start Docker Container
         run: |
           TARGET=${{steps.target.outputs.target}}
-          cid=$(docker ps -q --filter "name=${TARGET}")
-          if [[ ! -z "$cid" ]]; then docker stop $cid && docker rm $cid; fi
-          # check and remove exited container
-          cid=$(docker ps -a -q --filter "name=${TARGET}")
-          if [[ ! -z "$cid" ]]; then docker rm $cid; fi
-          docker run -tid -v ${{ inputs.model_cache_path }}:/root/.cache/huggingface/hub -v ${{ inputs.code_checkout_path }}:/root/llm-on-ray -e http_proxy=${{ inputs.http_proxy }} -e https_proxy=${{ inputs.https_proxy }} --name="${TARGET}" --hostname="${TARGET}-container" ${TARGET}:latest
+          code_checkout_path=${{ inputs.code_checkout_path }}
+          model_cache_path=${{ inputs.model_cache_path }}
+          # cid=$(docker ps -q --filter "name=${TARGET}")
+          # if [[ ! -z "$cid" ]]; then docker stop $cid && docker rm $cid; fi
+          # # check and remove exited container
+          # cid=$(docker ps -a -q --filter "name=${TARGET}")
+          # if [[ ! -z "$cid" ]]; then docker rm $cid; fi
+          # docker run -tid -v ${{ inputs.model_cache_path }}:/root/.cache/huggingface/hub -v ${{ inputs.code_checkout_path }}:/root/llm-on-ray -e http_proxy=${{ inputs.http_proxy }} -e https_proxy=${{ inputs.https_proxy }} --name="${TARGET}" --hostname="${TARGET}-container" ${TARGET}:latest
+          source docker.sh
+          clean_docker  TARGET=${TARGET} 
+          run_docker TARGET=${TARGET} code_checkout_path=${code_checkout_path} model_checkoutpath=${model_checkout_path} USE_PROXY=${"True"}
 
       - name: Start Ray Cluster
         run: |
           TARGET=${{steps.target.outputs.target}}
-          docker exec "${TARGET}" bash -c "./dev/scripts/start-ray-cluster.sh"
+          bash_command="./dev/scripts/start-ray-cluster.sh"
+          source docker.sh
+          docker_bash TARGET=${TARGET}  bash_command=${bash_command}
```

Comment:
It looks like you need to call this way: 
TARGET=${TARGET}  bash_command=${bash_command} docker_bash ?